pkg/resource/role/sdk.go

```diff
@@ -412,7 +412,15 @@ func (rm *resourceManager) sdkUpdate(
 			return nil, err
 		}
 	}
-	if !delta.DifferentExcept("Spec.Tags", "Spec.Policies", "Spec.InlinePolicies", "Spec.PermissionsBoundary") {
+
+	if delta.DifferentAt("Spec.AssumeRolePolicyDocument") {
+		err = rm.syncAssumeRolePolicies(ctx, desired)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	if !delta.DifferentExcept("Spec.Tags", "Spec.Policies", "Spec.InlinePolicies", "Spec.PermissionsBoundary", "Spec.AssumeRolePolicyDocument") {
```

Comment:
This file is auto generated. You can instead modify https://github.com/aws-controllers-k8s/iam-controller/blob/main/templates/hooks/role/sdk_update_pre_build_request.go.tpl and regenerate the controller

Reply:
Ah that'll do, I saw this template file last night and was wondering whether I'd missed something.

Thanks for clarifying that, I'll move the relevant change to here 😊